### DISCUSSION TITLE
隐语开源社区 首期“社区之星” 新鲜出炉！

### DISCUSSION THREAD

**Discussion by halosecret00** [secretflow/secretflow#246] [General]
“隐语开源社区”以安全开放为理念，致力于隐私计算技术研发及规模化应用的推进，解决行业数据安全、数据信任、数据孤岛问题。2022年是隐语开源的第一年，半年来的茁壮成长离不开大家的支持与信任，社区从0到1，越来越多小伙伴加入到隐语的共建中来，汇集了一批深度用户，他们互相学习探讨、交流技术，与此同时也发展出一批极具价值与影响力的开发者，以代码贡献、互动答疑、内容分享等不同形式影响着隐语社区。**为此，特别推出“隐语社区之星”评选活动，对促进社区发展，参与隐语社区共建的小伙伴给予正向激励！**

**如何成为社区之星**

社区将根据以下几点进行综合评判，包括，但不限于：pr代码贡献、完成社区发布任务集、bug反馈、issue提问、社群问答（提问/建议/答疑解惑）、学习分享（如成为慕课课代表、分享学习笔记）、社区活动参与度、公众号文章投稿等。

![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI34wkjZ1iaWtSmRf9zl0Pgb4HaDkUialuoiaiaIY6Y18LhcISq9sk5HvhvK4JbqEia7pH8mNjR9NTKI6bZA/640?wx_fmt=png)

ps：小伙伴们在github提交pr内容后，记得联系小助手哟～

**社区之星享受哪些权益🎉**

*   社区特别纪念礼品1份
    
*   “隐语社区之星”证书
    
*   成为“隐语开源开放日”特邀嘉宾
    

接下来向大家隆重介绍，隐语开源社区首期“社区之星”，有来自高校的莘莘学子，也有专注技术的行业精英，他们是在2022年隐语成长过程中不可或缺的一员，感谢小伙伴们在社区做出的贡献，一起来看看社区之星带来的寄语吧！

**🎖2022隐语开源社区「技术贡献之星」黄晓军**

![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI34wkjZ1iaWtSmRf9zl0Pgb4HDPHia5iaCrK19ybka6nsIkeUvSs3TA4RxOEtmZNaIzM7bkr60SUpdYFQ/640?wx_fmt=png)

**个人简介**

英特尔亚太研发有限公司高级云计算软件工程师。目前专注于云计算安全解决方案的开发、产品应用和客户支持。擅长领域包括同态加密、多方安全计算和可信执行环境在内的各种隐私保护技术的融合与应用，人工智能模型在Intel Xeon架构上的优化等。参与了多个开源项目的开发，如Intel Paillier Cryptosystem Library（IPCL），Confidential Computing Zoo（CCZoo）等，是Intel HE Toolkit在龙蜥社区CNCC SIG的maintainer。

pr贡献：https://github.com/secretflow/heu/pull/33

**社区之星寄语**

*   想法来源
    

同态加密技术在密文下进行运算，可以提供极高的数据安全保证，因此被广泛地应用于隐私保护计算、联邦学习和区块链等领域。其中，Paillier算法是一个广泛应用的加法半同态算法，隐语社区的HEU项目也有Paillier的相关实现。然而当时的HEU项目还没有一个专门针对Intel Xeon平台优化的Paillier实现，又恰好Intel今年刚开源了一个通过ISO认证的Paillier算法库IPCL，我和力科就想到可以把这个IPCL库集成进隐语的HEU项目，给用户提供多样化和更优性能的选择。

IPCL库使用了Intel IPP-Crypto库提供的AVX512IFMA指令集加速的能力，在最新的Intel Xeon平台上有优异的性能表现。同时在最新的IPCL v2.0.0的版本中，又引入了QAT加速器的加速能力，用户可以将部分工作负载卸载到QAT加速器进行运算，从而缓解CPU的运算压力。

*   技术细节
    

由于不同的Paillier库的大整数类型定义不尽相同，整个集成工作面临的最大挑战是如何在现有的架构上适配不同的大整数类型。我们和隐语团队的宇文同学一起讨论抽象出了接口层，隐语团队部分重构了HEU的代码，我们则负责将这些接口功能一一实现。其中有些要求的功能，比如大整数的逻辑运算功能等，在IPCL库中没有直接的实现，这就需要我们在适配层中自己来实现。最终经过通力合作，两个团队一起成功地将IPCL库引入了隐语的HEU项目。

**🎖2022隐语开源社区「技术贡献之星」段宇飞**

![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI34wkjZ1iaWtSmRf9zl0Pgb4HibCI9OBUCKE7kGY7sv42oicjic7ZQnHxOr0ZO5Y7WYd7j3pRMy1B7rFkw/640?wx_fmt=png)

**个人简介：**

目前就读于北京邮电大学网络空间安全学院，安全多方计算方向，已保送2023届清华大学网研院博士。本科阶段主要研习密码分析学方向相关内容，并在北京邮电大学天枢CTF战队担任密码手，拥有丰富的国内外各大CTF赛事经验，擅长数学和开发、热爱代码，自高中起参与各类信息学竞赛，本科阶段逐步积累开源项目经验，很多项目已在GitHub开源。

pr贡献：https://github.com/secretflow/spu/pull/53

**社区之星寄语：**

灵感其实是希望熟悉隐语框架的spu，感觉spu相比现在state-of-the-art的框架MP-SPDZ兼容性更强、开发者使用难度很低，就试着写了一个example，为了后面学习spu的底层打基础。  
在写之前，我是完全不懂spu的使用方法、jax编程、kmeans算法，这三者中最简单的竟是spu的使用方法。因为之前没接触过jax编程，也没了解过相关的思想，所以从头学以及写jax版本kmeans算法花费了一些时间。当然，之前没合作过大型项目，git的使用规范也花费了不少功夫。  
这个算法使用MPC技术实现了k-means算法。对于不同参与方输入的点，该算法可以“隐私保护地”找这些点的“聚点”，并向各个参与方返回点的标签和聚点坐标，但是不会泄露除此之外的任何信息。通过这个算法的代码可以看出来，开发者不需要明白任何spu的细节，甚至不需要对当前已有的非“隐私保护的”代码做改进，就可以用很小的开发代价来“升级”代码。

**🎖2022隐语开源社区「指点迷津之星」王艺团**

![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI34wkjZ1iaWtSmRf9zl0Pgb4Hudl50WeLSpuiblAt1Tj3f2w2L2DWSJzMfZJBtExBfHudWnrRVXp8haw/640?wx_fmt=png)

**个人简介：**

毕业于东北大学软件工程专业，目前就职于厦门掌讯信息科技有限公司，负责公司技术统筹规划和管理。从业以来，主要在烟草、食药监、公共安全等领域践行数据仓库、商业智能等大数据智能化技术实践；现阶段，服务于银行、清算组织等金融机构间的数据安全流通。

**社区之星寄语：**

初识隐语是在B站上看到隐语明密文混合计算的设计讲解，令人耳目一新。在此之前，已经调研了部分产品，然而商业产品太封闭，没有足够的API可以进行定制、优化；而开源产品，大多是“专科”，专注某个技术方向，尤其是以联邦学习居多。  
熟悉隐语过程中，时时在开放可扩展、分层分块等架构设计上有新的感受，从快速安装部署到教程实践，从应用API到底层协议实现，虽然不熟悉Python、C++，导致大部分时候只是读代码，但感觉每次都能学到新知识。友好积极的社区氛围也让我这个社区“社恐”逐步放开心态，喜欢上参与和帮助解答社群问题，既能深入思考、收获知识，也能体悟沟通、激发兴趣。  
隐私计算所包含的技术虽然不算是新概念，但其天生适用广泛的跨机构数据流通场景，既存在跨机构网络带宽约束下的性能提升、协议创新挑战，也存在跨机构信任有限带来的项目落地、数据开放范围的困境，希望接下来隐语在丰富模块、提升性能的同时，也可以看到依托隐语进行工程交付、项目管理方面的社区实践分享。

**🎖2022隐语开源社区「学习分享之星」杜林康**

**![图片](https://mmbiz.qpic.cn/mmbiz_png/rczZsXZbI34wkjZ1iaWtSmRf9zl0Pgb4HgIBD1C1xNWQmThmys32MboAuPOJBUu0ddiahgUnd2ZxnFQNayLPPWTg/640?wx_fmt=png)**

**个人简介：**

目前就读于浙江大学控制科学与工程学院网络空间安全专业，浙江大学-德国亥姆霍兹信息安全中心（CISPA）联合培养博士。主要研究兴趣在数据隐私保护及可信机器学习等交叉领域研究。开展了差分隐私、分布式机器学习、深度强化学习等领域研究。目前已在ACM CCS, IEEE TSP, IEEE CDC, USENIX RAID等期刊、会议发表论文5篇，受理授权专利3项，另有多篇成果在审。

**社区之星寄语：**

最初了解到隐语是从导师转发的隐语框架开源发布会推文，作为隐私计算的研究者，十分兴奋蚂蚁集团可以推出一款面向生产的开源计算框架。虽然隐私计算方面的研究保守估计已经有5-10年的时间，但是成熟的可商用的开源框架还十分稀少。笔者主要关注差分隐私领域的开源框架，比较有名的例如Google differential-privacy。  
隐语的慕课内几乎容涵盖了隐私计算各个子领域的基础知识，并且每一节课都是该领域一线的老师/技术大牛讲解，在讲解PPT内容的同时，能够很好地补充领域最新知识，获益匪浅。差分隐私部分的主讲人Tianhao Wang教授在博士期间发表了很多关于本地差分隐私的高质量研究工作，也启发了我后续发表在CCS 21的研究工作。

期待有更多从事隐私计算工作的同行加入隐语开源框架的开发和维护，希望隐语团队可以更多地组织一些基于隐语框架开发的算法设计大赛，丰富隐语开源社区文化。

* * *

最后，期待更多开发者以自己喜欢、擅长的方式加入到隐语社区的共建，小伙伴们也可以通过自荐的方式参与到下一期的评选，与社区共同探索出更多的共建方向～❤️打造出一个充满活动力的开源社区！

**Comment by yeshuchen323** [comment]
感谢社区开发者们的贡献❤
